### PR TITLE
Many handlers perf2.1

### PR DESCRIPTION
**Pull request by Macroz**
Replaces #3313 

Continues work in #3305 

Includes:
- Makes sure handled applications list is fetched lazily so correct amount of work is done when a limit is given (e.g. only 50).
- Logs test data creation speed to be able to easily follow it. This gives an indication of general REMS performance.
- Emails for one application were sent one at a time. E.g., with 12 handlers there would 12 separate calls to save the email into the outbox. Now this is done with a single call, which helps the 100 handler "worst-case scenario". Tests needed updating to use the new function.
- Replaced `mapcat` with other code to make it faster (micro-optimization).
- The main problem for the performance of the `/api/applications/handled` is actually the verification of the return Schema. If there are thousands of applications, there is a lot of data and half of the time is just spent verifying that it matches our Schema. We could investigate if we can leave the checking out in some cases (e.g., when we know there are a lot of applications). See [issue](https://github.com/orgs/CSCfi/projects/13/views/1?pane=issue&itemId=75940840)

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue

## Backwards compatibility
- [x] API is backwards compatible or completely new

## Documentation
- [ ] Update changelog if necessary  (no need for a new one as the last improvement is already mentioned)

## Testing
- [ ] Complex logic is unit tested
- [x] Valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [x] New tasks are created for pending or remaining tasks (see [issue](https://github.com/orgs/CSCfi/projects/13/views/1?pane=issue&itemId=75940840))

### CODE REVIEW

File: src/clj/rems/api/applications.clj [hunk 1]
```diff
@@ -176,15 +184,14 @@
     (GET "/handled" []
       :summary "Get all applications that the current user no more needs to act on."
       :roles #{:logged-in}
+      ;; XXX: checking this schema can be mighty slow (thousands of applications to check individually)
```
Comment (aatkin) [inline src/clj/rems/api/applications.clj:187]: there seems to be a newer version of compojure-api released: https://github.com/metosin/compojure-api/blob/master/CHANGELOG.md#200-alpha33-2024-04-30

though testing locally, e.g. resources api with performance data is still as slow as with older version.

**Review by aatkin — APPROVED** [review]
going ever faster! :+1:

File: src/clj/rems/email/core.clj [hunk 2]
```diff
@@ -30,35 +30,43 @@
...
-                              (template/reviewer-reminder-email lang reviewer apps))))
-                     (remove nil?))]
-    (enqueue-email! email)))
+  (let [deadline (-> (time/now) (.plus ^Period (:email-retry-period env)))]
+    (->> (for [email (->> (applications/get-users-with-role :reviewer)
+                          (map users/get-user)
+                          (map (fn [reviewer]
+                                 (let [lang (:language (user-settings/get-user-settings (:userid reviewer)))
+                                       apps (->> (todos/get-todos (:userid reviewer))
+                                                 (map #(= :waiting-for-your-review (:application/todo %))))]
```
[the remark this answers is not in the document]
Comment (Macroz) [reply src/clj/rems/email/core.clj:63]: I can see if I can refactor this. `keep` would work here but there's more.

I think the tests are in `rems.api.test-email`, e.g., `test-send-reviewer-reminder`.

File: src/clj/rems/api/applications.clj (hunk 1 above)
Comment (Macroz) [reply src/clj/rems/api/applications.clj:187]: We should keep an eye on it an upgrade once there is a good "2.0" or so.

**Pull request merged**